modules/database/funcs.go

```diff
@@ -262,9 +258,44 @@ func (d *Database) WaitForDBCreated(
 	return d.WaitForDBCreatedWithTimeout(ctx, h, time.Second*5)
 }
 
-//
+// Get MariaDB instance
+func (d *Database) getDBInstance(
+	ctx context.Context,
+	h *helper.Helper,
+) error {
+	namespace := d.namespace
+	if namespace == "" {
+		namespace = h.GetBeforeObject().GetNamespace()
+	}
+	// Fetch the MariaDB instance
+	dbInstance := &mariadbv1.MariaDB{}
+	err := h.GetClient().Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: namespace,
+			Name:      "openstack",
+		},
+		dbInstance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
```

Comment:
It seems the code in the if branch is the same as the code outside of the if branch so you can merge them.